packages/engine/src/simulation/package/output/packages/json_state/mod.rs

```diff
@@ -76,12 +73,19 @@ impl GetWorkerSimStartMsg for JsonState {
 impl Package for JsonState {
     async fn run(&mut self, state: Arc<State>, _context: Arc<Context>) -> Result<Output> {
         let state = state.read()?;
-        let agent_states: std::result::Result<Vec<_>, crate::datastore::error::Error> = state
+        // Output packages can't reload state batches, since they only
+        // have read access to state, but reloading would mean
+        // mutating the loaded data.
+        let agent_states: crate::datastore::Result<Vec<_>> = state
             .agent_pool()
             .batches_iter()
             .zip(state.message_pool().batches_iter())
-            .map(|(agent_batch, message_batch)| {
-                (agent_batch.record_batch(), message_batch.record_batch())
+            .map(|(agent_group, message_batch)| {
+                let _ = (&state, &_context);
```

Comment:
What's going on here? Is it to satisfy the changes to closures and such with the new version of rust? If so can we add a comment about what this is actually doing? Also lets rename `_context` to `context` then.

Reply:
Yeah it's about the changes to closures

Reply:
Not needed, removed